packages/kbn-optimizer/src/worker/webpack.config.ts

```diff
@@ -64,6 +64,15 @@ export function getWebpackConfig(bundle: Bundle, bundleRefs: BundleRefs, worker:
 
     optimization: {
       noEmitOnErrors: true,
+      splitChunks: {
+        minSize: 1,
```

Comment:
Why set the `minSize` to 1 byte?

Reply:
`minSize` is the amount of size a module must be in order to be deduplicated, and we want every module to be deduplicated, and my experiments seemed to indicate that `0` was treated as `false`-y.

Reply:
Wouldn't changing this default from 20kb possibly result in extremely small chunks being created? It seems like the tradeoff is between download size and the overhead of possibly another request to fetch the chunk.

Reply:
My understanding is that this won't trigger new chunk creation, just make chunks a little smaller on average because there is less duplication and instead cause one chunk to depend on another chunk because it already has a small module which could have been duplicated and placed in both chunks instead.

I think the drawback is that the front-end might need to download two chunks rather than one just because it needs a tiny module from the second chunk. My assumption though is that nearly all chunks in a plugin will be required anyway, but I haven't validated that.

The signal that makes me confident this is _better_™ is the fact that the distributable file count is down in both the default and oss distributable, so I can't imagine that this would cause more or too many network requests.

Reply:
Compared the files in both distributables and it does seem that in some cases more chunks are created, and in some cases there are less... https://gist.github.com/spalger/5b76aed75991daa58cbdc8d323877940

Plugins that got new chunks, need to investigate why:
`triggers_actions_ui`: +7
`security`: +8
`ml`: +2
`infra`: +4
`apm`: +2

Plugins that had one or two bundles removed:
`advanced_settings`, `beats_management`, `console`, `cross_cluster_replication`, `dashboard`, `file_upload`, `graph`, `grokdebugger`, `home`, `index_lifecycle_management`, `index_management`, `index_pattern_management`, `ingest_pipelines`, `kibana_react`, `lens`, `license_management`, `logstash`, `maps`, `maps_legacy`, `monitoring`, `observability`, `region_map`, `remote_clusters`, `rollup`, `saved_objects_management`, `searchprofiler`, `security`, `tile_map`, `timelion`, `transform`, `upgrade_assistant`, `vis_default_editor`, `vis_type_table`, `vis_type_tagcloud`, `vis_type_timeseries`, `vis_type_vega`, `vis_type_vislib`, `visualize`, `canvas`, `ingest_manager`, `security_solution`

This seems like a reasonable tradeoff too me, but I'm going to see if I can figure out why plugins like `security` got so many new chunks.

Reply:
@tylersmalley So, you're right, the `maxSize` seems to allow tiny chunks to be created and plugins which create lots of chunks and have lots of async imports have a decent chance of creating very small chunks less than 20kb. That said, I think the increase in chunks is fine.

The `security` bundle includes several apps/screens and each is broken into a handful of chunks
 - login page: 4 chunks
 - user editor: 8 chunks
 - role editor: 6 chunks, 2 shared with the user editor
 - api keys editor: 2 chunks, 2 shared with user editor
 - etc.

Since security isn't consumed primarily as a single app, it is consumed as a group of screens in different places throughout Kibana, I think it makes sense that it's chunked into fairly small chunks. Additionally, this is controlled by the security team and by how often they use patterns like this https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/public/management/roles/roles_management_app.tsx#L38-L54

The `triggers_actions_ui` does async imports for several editor UIs for different trigger types, and these async imports are only loaded when the different trigger type editors are revealed (using `React.lazy()`).

These are the two big examples of extra chunk creation and I think it's ultimately fine, and also probably better that the overall size of these are reduced.

I'm not passionate about this though, do you have thoughts about what might be a better choice?

Reply:
Alright, looking more closely at this and doing some pair experiments we've decided that we're going to [record the count of async chunks](https://github.com/elastic/kibana/pull/80606) and try both `minSize: 10000` and `minSize: 1000`, and if the metrics improve dramatically with one over the other we'll go that route.